### PR TITLE
Add Message Validation

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add validation to check if participant is messageable
* Add model specs to check if participant AND if arm implements messaging

[Finished: #107436990]

### CODE REVIEW

File: app/models/message.rb [hunk 1]
```diff
@@ -30,11 +32,26 @@ def render_body
...
+      .current_group
+      .arm
+      .bit_core_tools
+      .find_by_type("Tools::Messages")
+  end
+
+  def participant_is_messageable
+    unless messaging_tool?
+      errors
+        .add(:base, "We're sorry, but this participant can't receive emails.")
```
Comment (michaelwehrley) [inline app/models/message.rb]: @cbrenner04 can you confirm that this language is okay?  If not, can you provide the correct language. Thanks.

Comment (cbrenner04) [inline app/models/message.rb]: I think "We're sorry, but this participant does not have access to the messaging tool." is more informative.

Comment (michaelwehrley) [inline app/models/message.rb]: @cbrenner04 thanks you.

File: app/models/message.rb [hunk 2]
```diff
@@ -30,11 +32,27 @@ def render_body
...
 
-  def populate_sent_at
-    self.sent_at = Time.new
-  end
-
   def create_delivered_messages
     recipient.received_messages.create(message_id: id)
   end
+
+  def messaging_tool?
```
Comment (ericcf) [inline app/models/message.rb]: Why the `?` in the name. Usually the convention is for methods that return booleans to have a `?`.

Comment (michaelwehrley) [inline app/models/message.rb]: bless it, after refactoring I removed the `false`.  :lame.

Comment (michaelwehrley) [inline app/models/message.rb]: well because it returns truthy or falsey.

Comment (michaelwehrley) [inline app/models/message.rb]: but I can fix that.

Comment (ericcf) [inline app/models/message.rb]: All Ruby methods return truthy or falsy :-)

Comment (michaelwehrley) [inline app/models/message.rb]: LOL, that is ridiculously truthy.

**Pull request merged**